### PR TITLE
Checkpoints arrow customization

### PR DESCRIPTION
**Pull request by FileEX**
As I said in #3523  ,I added the ability to customize the target arrow in checkpoint markers. You can change the color of the arrow and its size.

**Syntax**
``bool setMarkerTargetArrowProperties(element marker [, int r = 255, int g = 64, int b = 64, int a = 255, float size = markerSize * 0.625 ] )`` - If we do not pass arguments, the default properties will be restored (something like restart).

``r, g, b, a, size = getMarkerTargetArrowProperties(element marker)``

Screenshot_1

### CODE REVIEW

**Comment by Proxy-99** [comment]
is it possible to support OOP too
something like 
`marker:setTargetColor(255,0,0,255)`
`marker:setTargetSize(2)`
get
`marker.targetColor`
`marker.targetSize`

File: Client/game_sa/CCheckpointSA.h [hunk 1]
```diff
@@ -14,6 +14,12 @@
 #include <CVector.h>
 #include <game/CCheckpoint.h>
 
+#define HOOKPOS_CCheckpoint__Render  0x725E56
+#define HOOKSIZE_CCheckpoint__Render 0x5
+static const constexpr std::uint32_t RETURN_CCheckpoint__Render = 0x725E5B;
```
Comment (TracerDS) [inline Client/game_sa/CCheckpointSA.h]: `const` is unnecessary here

File: Client/mods/deathmatch/logic/CClientCheckpoint.cpp [hunk 2]
```diff
@@ -385,3 +390,20 @@ void CClientCheckpoint::ReCreateWithSameIdentifier()
...
+
+void CClientCheckpoint::SetTargetArrowProperties(const SColor& arrowColor, float size) noexcept
+{
+    if (m_TargetArrowColor != arrowColor || m_TargetArrowSize != size)
+    {
+        m_TargetArrowColor = arrowColor;
+        m_TargetArrowSize = size;
+
+        ApplyCheckpointTargetArrowProperties();
+    }
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/CClientCheckpoint.cpp]: early return?

File: Client/mods/deathmatch/logic/rpc/CMarkerRPCs.cpp [hunk 3]
```diff
@@ -137,4 +138,22 @@ void CMarkerRPCs::SetMarkerIcon(CClientEntity* pSource, NetBitStreamInterface& b
...
+        CClientMarker* marker = m_pMarkerManager->Get(pSource->GetID());
+        if (!marker)
+            return;
+
+        CClientCheckpoint* checkpoint = marker->GetCheckpoint();
+        if (!checkpoint)
+            return;
+
+        checkpoint->SetTargetArrowProperties(color, size);
+    }
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/rpc/CMarkerRPCs.cpp:158]: early return?

File: Server/mods/deathmatch/logic/CMarker.cpp [hunk 4]
```diff
@@ -301,6 +305,23 @@ void CMarker::SetIcon(unsigned char ucIcon)
...
+        m_TargetArrowSize = size;
+
+        CBitStream BitStream;
+        BitStream.pBitStream->Write(color.R);
+        BitStream.pBitStream->Write(color.G);
+        BitStream.pBitStream->Write(color.B);
+        BitStream.pBitStream->Write(color.A);
+        BitStream.pBitStream->Write(size);
+        BroadcastOnlyVisible(CElementRPCPacket(this, SET_MARKER_TARGET_ARROW_PROPERTIES, *BitStream.pBitStream));
+    }
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CMarker.cpp]: early return?

**Review by TracerDS — CHANGES_REQUESTED** [review]

**Comment by TracerDS** [comment]
> `r, g, b, a, size = getMarkerTargetArrowProperties(element marker)`

What do you think about `getMarkerTargetArrowSize` and `getMarkerTargetArrowColor`?

**Comment by FileEX** [comment]
I think there is really no need to create two separate functions for this, because it is very simple and there are only 2 arguments - color and size.

**Comment by Proxy-99** [comment]
> I think there is really no need to create two separate functions for this, because it is very simple and there are only 2 arguments - color and size.

you can use it for the oop syntax tho I think better to separate them for better readability

**Comment by FileEX** [comment]
OOP is not mandatory.  If someone wants to make OOP or setMarkerTargetArrow / setMarkerTargetArrowSize versions, feel free to do so

**Pull request merged**